src/java/org/apache/cassandra/index/sai/disk/vector/CompactionGraph.java

```diff
@@ -537,4 +629,64 @@ public InsertionResult(long bytesUsed)
             this(bytesUsed, null, null);
         }
     }
+
+    private static class OnDiskVectorValues implements RandomAccessVectorValues, AutoCloseable
+    {
+        private final RandomAccessReader reader;
+        private final int dimension;
+        private final long vectorSize;
+
+        public OnDiskVectorValues(File vectorsByOrdinalTmpFile, int dimension)
+        {
+            this.reader = RandomAccessReader.open(vectorsByOrdinalTmpFile);
+            this.dimension = dimension;
+            this.vectorSize = (long) dimension * Float.BYTES;
+        }
+
+        @Override
+        public int size()
+        {
+            return (int) (reader.length() / vectorSize);
+        }
+
+        @Override
+        public int dimension()
+        {
+            return dimension;
+        }
+
+        @Override
+        public VectorFloat<?> getVector(int i)
+        {
+            try
+            {
+                reader.seek(i * vectorSize);
+                var vector = new float[dimension];
+                reader.readFully(vector);
+                return vts.createFloatVector(vector);
+            }
+            catch (IOException e)
+            {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public boolean isValueShared()
+        {
+            return false;
+        }
+
+        @Override
+        public RandomAccessVectorValues copy()
+        {
+            return this;
+        }
```

Comment:
Can we really safely copy this?
This class has some mutable state in it in form of the `reader` and its file position.
If I make a copy by calling `copy()` and then use the returned instance in a different thread concurrently.... it will do weird things. 


If it's never called / needed, maybe just throw `UnsupportedOperationException` instead?